### ISSUE TITLE
Front-end Tests not executed during pipelines build

### ISSUE THREAD

**Issue by aweingarten** [acquia/blt#1631]
When I run a pipelines build on the head of 8.9.x the following hook does not fire
```
target-hooks:
  frontend-test:
    dir: '${docroot}/themes/custom/freerange'
    command: 'yarn install && npm run pa11y'
```

I think a regression was introduced when the tests were refactored to provide a webserver in CI.

**Comment by aweingarten** [comment]
@grasmash I'm still seeing issues

```
2 scenarios (2 passed)
4 steps (4 passed)
0m1.10s (29.70Mb)
[info] Killing all processes on port '4444'...
[info] Killing all processing containing string 'bin/phantomjs'...
[info] Killing all processing containing string 'runserver'...
[info] Killing all processes on port '8888'...
> tests:phpunit
[Filesystem\FilesystemStack] mkdir ["/mnt/tmp/local.prod/source/reports/phpunit"]
[Filesystem\FilesystemStack] touch ["/mnt/tmp/local.prod/source/reports/phpunit/results.xml"]
[Testing\PHPUnit] Running PHPUnit  --log-junit /mnt/tmp/local.prod/source/reports/phpunit/results.xml .
PHPUnit 4.8.35 by Sebastian Bergmann and contributors.

.

Time: 48 ms, Memory: 5.25MB

OK (1 test, 1 assertion)
> tests:security-updates
There are no outstanding security updates for Drupal projects.
> frontend:test
Executing frontend-test target hook...
[ExecStack] yarn install && npm run pa11y
yarn install v0.24.5
[1/4] Resolving packages...
[2/4] Fetching packages...
warning fsevents@1.0.14: The platform "linux" is incompatible with this module.
info "fsevents@1.0.14" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
$ find node_modules/ -name '*.info' -type f -delete
Done in 128.49s.

> freerange@0.1.0 pa11y /mnt/tmp/local.prod/source/docroot/themes/custom/freerange
> gulp pa11y-test

[00:15:26] Using gulpfile /mnt/tmp/local.prod/source/docroot/themes/custom/freerange/gulpfile.js
[00:15:26] Starting 'pa11y-test'...
[00:15:26] Accessibility Audit starts
PhantomJS browser created
PhantomJS browser created
PhantomJS page created
PhantomJS page created
Opening "http://127.0.0.1:8888" in PhantomJS
Opening "http://127.0.0.1:8888/themes/custom/freerange/styleguide" in PhantomJS
PhantomJS page closed
PhantomJS browser exited
[00:15:27] 'pa11y-test' errored after 156 ms
[00:15:27] Error in plugin 'pa11y'
Message:
    Error opening url "http://127.0.0.1:8888" : Connection refused
PhantomJS page closed
PhantomJS browser exited
[00:15:27] 'pa11y-test' errored after 158 ms
[00:15:27] Error in plugin 'pa11y'
Message:
    Error opening url "http://127.0.0.1:8888/themes/custom/freerange/styleguide" : Connection refused
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! freerange@0.1.0 pa11y: `gulp pa11y-test`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the freerange@0.1.0 pa11y script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/local/.npm/_logs/2017-06-13T00_15_27_583Z-debug.log
[ExecStack]  Exit code 1 
[error]  Executing target-hook frontend-test failed. 
pipelines_kill_background_processes
pstree -p $$ | grep -E -o '([0-9]+)' | grep -v "^$$\$" | sort -k2 -rn
Killing background jobs
```

**Comment by grasmash** [comment]
#1645

**Comment by loopy3025** [comment]
We're seeing the same thing. I'm trying to include a pa11y test during setup and it looks like blt is completely skipping source:build:frontend-test. 

As an alternative, I've tried adding the test during the pipelines setup-env script instead, but it doesn't seem to want to connect to localhost to do the actual testing, so that's probably not gonna work. :(

**Comment by mikemadison13** [comment]
have you defined the command hook? https://github.com/Drupal4Gov/Drupal-GovCon-2017/blob/develop/blt/blt.yml#L22

**Comment by loopy3025** [comment]
Yeah...
```
command-hooks:
  frontend-test:
    dir: '${docroot}/themes/custom/<our-theme-name>'
    command: 'npm run accessibility ci'
```
We're running blt 13.5.2. Perhaps something has changed since this issue? It seems rather old.

**Comment by loopy3025** [comment]
I didn't add the frontend-reqs or frontend-assets command hooks, however. I assumed that it'd use whatever defaults exist if I didn't. I shouldn't assume that. I'll have a go.

**Comment by loopy3025** [comment]
Although I'm not sure what to put in there. We're already running npm install during frontend-setup under target-hooks.

**Comment by mikemadison13** [comment]
i usually would define an "npm run tests" or something like that command that would execute. see https://github.com/Drupal4Gov/Drupal-GovCon-2017/blob/develop/docroot/themes/custom/twentynineteen/package.json#L57

let me run a pipeline today and see what i can see on one of my projects.

**Comment by loopy3025** [comment]
Hey, Mike. Yes, my script does call `npm run tests`. I just have the script so it can do some environment detection first.

**Comment by loopy3025** [comment]
Aha. I think I figured it out @mikemadison13 . I was looking at it and I saw that pipelines explcitly referenced setup_env, so I looked in the vendor folder at the commands available and saw a tests command in there. Sure enough, Pipelines doesn't automatically run the tests. I had to add the following to my pipelines setup-env script:

```
source ${BLT_DIR}/scripts/pipelines/tests
```
And it finally did run the command I put under command-hooks:frontend-tests.

It is, unfortunately, now giving me the same error it was throwing when I had my command directly under the pipelines build:

```
	> oursite@1.0.0 tests /mnt/tmp/local.prod/source/docroot/themes/custom/our-theme
	> pa11y-ci "https://127.0.0.1/"
	
	(node:5242) ExperimentalWarning: The fs.promises API is experimental
	
	Running Pa11y on 1 URLs:
	
	 > https://127.0.0.1/- Failed to run
	
	Errors in https://127.0.0.1/:
	
	 • Error: net::ERR_CONNECTION_REFUSED at
	   https://127.0.0.1/
	
	 0/1 URLs passed

```

But I suspect this falls out of the purview of BLT and should probably be directed at my Acquia rep, unless you have any quick ideas?

I appreciate your help.
